src/main/java/hudson/remoting/AbstractByteBufferCommandTransport.java

```diff
@@ -0,0 +1,286 @@
+                        if (ChunkHeader.isLast(readFrameHeader)) {
+                            readState = READ_STATE_COMMAND_READY;
+                        } else {
+                            readState = READ_STATE_NEED_HEADER;
+                        }
+                    }
+                    break;
+                case READ_STATE_COMMAND_READY:
+                    // we have read in a command
+                    readCommandIndex++;
+                    if (readCommandIndex > readCommandSizes.length) {
+                        readCommandSizes = Arrays.copyOf(readCommandSizes, readCommandSizes.length * 2);
+                    }
+                    readCommandSizes[readCommandIndex] = 0;
+                    readState = READ_STATE_NEED_HEADER;
+                    if (receiver != null) {
+                        processCommand();
+                    }
+                    break;
+            }
+        }
+    }
+
+    private void processCommand() throws IOException {
+        try {
+            FastByteBufferQueueInputStream is = new FastByteBufferQueueInputStream(receiveQueue, readCommandSizes[0]);
+            try {
+                ObjectInputStreamEx ois = new ObjectInputStreamEx(is, channel.baseClassLoader, channel.classFilter);
+                receiver.handle(Command.readFrom(channel, ois));
+            } catch (IOException e1) {
+                LOGGER.log(Level.WARNING, "Failed to construct Command", e1);
+            } catch (ClassNotFoundException e11) {
+                LOGGER.log(Level.WARNING, "Failed to construct Command", e11);
+            } finally {
+                int available = is.available();
+                if (available > 0) {
+                    is.skip(available);
+                }
+                IOUtils.closeQuietly(is);
+            }
+        } finally {
+            if (readCommandIndex == 1) {
+                if (readCommandSizes.length > 16) {
+                    // de-allocate the grown array
+                    int temp = readCommandSizes[1];
+                    readCommandSizes = new int[16];
+                    readCommandSizes[0] = temp;
+                } else {
+                    readCommandSizes[0] = readCommandSizes[1];
+                }
+                readCommandIndex = 0;
+            } else {
+                // we have processed position 0
+                // we now want position 1 to be the new position 0
+                // position readCommandIndex has the readState data length, so need to keep that
+                System.arraycopy(readCommandSizes, 1, readCommandSizes, 0, readCommandIndex);
+                readCommandIndex--;
+            }
+        }
+    }
+
+    /**
+     * Set the frame size.
+     *
+     * @param transportFrameSize the new frame size.
+     */
+    public void setFrameSize(int transportFrameSize) {
+        if (transportFrameSize <= 0 || transportFrameSize >= Short.MAX_VALUE) {
+            throw new IllegalArgumentException();
```

Comment:
No description. Also makes sense to mention restrictions and the potential exception in Javadoc